### ISSUE TITLE
Windows 10 display scaling not handled properly

### ISSUE THREAD

**Issue by liujed** [TigerVNC/tigervnc#494]
I have the server running on a Windows 10 machine with two monitors of different resolutions and pixel densities. The Win10 display settings is configured to scale the left monitor by 250%, and the right one by 125%. When I connect with a Linux client, the mouse cursor on the client appears in a different location than what is shown on the server. When I click the mouse on the client, the click registers according to the mouse cursor's location on the server.

**Comment by CendioOssman** [comment]
High DPI monitors is currently not supported, but I'll leave this issue as a tracker for that feature.

Also see #369 for the client side of things.

**Comment by dhpriest** [comment]
I've confirmed this issue affects single-monitor setups as well under v1.6.0 on Windows Server 2008R2 and Windows 10. I've not yet tested other Windows versions or Linux systems. Differences in DPI between client/server don't seem to matter but the server-side DPI scaling that can be changed through Win+R ( dpiscaling ) on Windows systems causes disparity between the client/server mouse cursor positions. TightVNC has the same problem. Should this be a separate issue?

**Comment by CendioOssman** [comment]
I think that can be included here.

**Comment by ramcq** [comment]
I had the input not matching the screen scaling issue. I found that the workaround described in https://github.com/TigerVNC/tigervnc/issues/480 actually works for the server as well. I used the Compatibility tab in Properties on winvnc4.exe and pressed "Change settings for all users" and set the high DPI scaling to "Application", then restarted the service. I assume setting it to "Application" just stops Windows from performing any translation on the input or output - that means that the remote client input matches what's seen on screen. I guess there is a capability one can export to tell Windows that the application handles it.

**Comment by ramcq** [comment]
(I think you could set dpiAware in the manifest, or use equivalent API calls, to turn off DPI scaling that Windows applies to the server service - https://msdn.microsoft.com/en-us/library/windows/desktop/aa374191(v=vs.85).aspx )

**Comment by ramcq** [comment]
Yes, setting the compatibility options on the .exe file and restarting the service worked for me.
